src/thunderbots/software/ai/hl/stp/action/stop_action.cpp

```diff
@@ -0,0 +1,27 @@
+#include "ai/hl/stp/action/stop_action.h"
+
+#include "ai/intent/stop_intent.h"
+#include "geom/angle.h"
+#include "geom/util.h"
+#include "shared/constants.h"
+
+StopAction::StopAction() : Action() {}
+
+std::unique_ptr<Intent> StopAction::updateStateAndGetNextIntent(const Robot& robot,
+                                                                bool coast)
+{
+    // update the parameters stored by this action
+    this->robot = robot;
+    this->coast = coast;
+
+    return getNextIntent();
+}
+
+std::unique_ptr<Intent> StopAction::calculateNextIntent(
+    intent_coroutine::push_type& yield)
+{
+    do
+    {
+        yield(std::make_unique<StopIntent>(robot->id(), coast, 0));
```

Comment:
Hm should Stops get the highest priority? @MathewMacDougall

Reply:
I would think so. If this is used when a Ref calls for a halt we would want to stop immediately.

Reply:
Yes, but the priority system isn't being used yet, and I'm not quite sure how it will be. @jonl112 ?

Reply:
currently have it set to 0, we will have to update all of them anyway